dissect/target/plugins/os/unix/linux/network.py

```diff
@@ -0,0 +1,324 @@
+                    continue
+
+                for ip_version in ["ipv4", "ipv6"]:
+                    ip_section: dict[str, str] = config.get(ip_version, {})
+                    for key, value in ip_section.items():
+                        self._parse_ip_section_key(key, value, context, ip_version)
+
+                context.name = common_section.get("interface-name")
+                context.mac_address = sub_type.get("mac-address")
+                context.uuid = common_section.get("uuid")
+                context.source = str(connection_file_path)
+                context.last_connected = self._parse_lastconnected(common_section.get("timestamp", ""))
+
+                connections.append(context)
+
+            except Exception as e:
+                self._target.log.warning("Error parsing network config file %s: %s", connection_file_path, e)
+
+        for connection in connections:
+            uuid = vlan_id_by_interface.get(context.uuid) if context.uuid else None
+            name = vlan_id_by_interface.get(connection.name) if connection.name else None
+            if vlan_ids := (name or uuid):
+                connection.vlan.update(vlan_ids)
+            yield connection.to_record()
+
+    def _parse_route(self, route: str) -> ip_address | None:
+        """Parse a route and return gateway IP address."""
+        if (elements := route.split(",")) and len(elements) > 1:
+            return ip_address(elements[1])
+
+        return None
+
+    def _parse_lastconnected(self, last_connected: str) -> datetime | None:
+        """Parse last connected timestamp."""
+        if not last_connected:
+            return None
+
+        return datetime.fromtimestamp(int(last_connected), timezone.utc)
+
+    def _parse_ip_section_key(
+        self, key: str, value: str, context: ParserContext, ip_version: Literal["ipv4", "ipv6"]
+    ) -> None:
+        # nmcli inserts a trailling semicolon
+        if key == "dns" and (stripped := value.rstrip(";")):
+            context.dns.update({ip_address(addr) for addr in stripped.split(";")})
+        elif key.startswith("address") and (trimmed := value.strip()):
+            # Undocumented: single gateway on address line. Observed when running:
+            # nmcli connection add type ethernet ... ip4 192.168.2.138/24 gw4 192.168.2.1
+            ip, *gateway = trimmed.split(",", 1)
+            context.ip_interfaces.add(ip_interface(ip))
+            if gateway:
+                context.gateways.add(ip_address(gateway[0]))
+        elif key.startswith("gateway") and value:
+            context.gateways.add(ip_address(value))
+        elif key == "method" and ip_version == "ipv4":
+            context.dhcp_ipv4 = value == "auto"
+        elif key == "method" and ip_version == "ipv6":
+            context.dhcp_ipv6 = value == "auto"
+        elif key.startswith("route"):
+            if gateway := self._parse_route(value):
+                context.gateways.add(gateway)
```

Comment:
Looking at this again, I actually prefer the nested if because they have different purposes.

The outer `if` switches on the key type, and the inner `if` is part of the action of handling a certain configuration value. This way, we can scan more easily over the different key types.

I will make the outer clauses consistent.